### PR TITLE
[3.6] Set writeConcern on DBServers

### PR DESCRIPTION
**Pull request by goedderz**
Backport of https://github.com/arangodb/arangodb/pull/12366

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11365/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11409/

**Comment by goedderz** [comment]
image

File: arangod/VocBase/Methods/Collections.cpp [hunk 1]
```diff
@@ -289,18 +289,50 @@ Result Collections::create(TRI_vocbase_t& vocbase,
...
+        TRI_ASSERT(isLocalCollection);
+        break;
+      case ServerState::ROLE_DBSERVER:
+        TRI_ASSERT(isLocalCollection == isSystemName);
+        break;
+      case ServerState::ROLE_COORDINATOR:
+        TRI_ASSERT(!isLocalCollection);
+        break;
+      case ServerState::ROLE_AGENT:
+        TRI_ASSERT(isLocalCollection && isSystemName);
```
Comment (neunhoef) [inline arangod/VocBase/Methods/Collections.cpp:318]: Is this going to work? I think an agent has collections named `log` and `compact` and `config`, which are not system collections. They will be local, but will not have a system name. They are created only once pretty early on, maybe it works for this reason.

Comment (goedderz) [reply arangod/VocBase/Methods/Collections.cpp:318]: Ah. I wrongly assumed that the agents' collections would start with an underscore.

The tests are successful, but I better fix that assertion anyway. Thanks!

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11423/

File: arangod/VocBase/Methods/Collections.cpp (hunk 1 above)
Comment (goedderz) [reply arangod/VocBase/Methods/Collections.cpp]: I've fixed the assertion. And i also verified that this code is just not executed on agents to create those collections. They are created in `arangod/Agency/State.cpp`, in `State::createCollections()`, which directly calls `TRI_vocbase_t::createCollection()`.

**Comment by goedderz** [comment]
image

File: arangod/VocBase/Methods/Collections.cpp [hunk 2]
```diff
@@ -289,18 +289,50 @@ Result Collections::create(TRI_vocbase_t& vocbase,
...
+    // collection (as seen on a Coordinator), nor a shard (on a DBServer).
+    bool const isLocalCollection =
+        (!ServerState::instance()->isCoordinator() &&
+         Helper::stringUInt64(info.properties.get(StaticStrings::DataSourcePlanId)) == 0);
+
+    bool const isSystemName = TRI_vocbase_t::IsSystemName(info.name);
+
+    // All collections on a single server should be local collections.
+    // A Coordinator should never have local collections.
+    // On an Agent, all collections should be both local and system collections.
```
Comment (goedderz) [inline arangod/VocBase/Methods/Collections.cpp:303]:
```suggestion
    // On an Agent, all collections should be local collections.
```

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**